### PR TITLE
Tskhwk 65 frontend implement login page

### PR DESCRIPTION
**Pull request by fernandorochaworld**
Created the login page and added structure needed to the login feature.

The Setting page is still included and it will be improved in other ticket.

image

### CODE REVIEW

File: client/src/components/PrivateRoute.tsx [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import { Navigate, useLocation } from 'react-router-dom';
+import { useAuth } from '@context/useAuth';
```
Comment (ronaldomaiacorrea) [inline client/src/components/PrivateRoute.tsx]: Let's move this useAuth to hooks **folder**, not **context** one, I think is more appropriated.

File: client/src/features/settings/Settings.tsx [hunk 2]
```diff
@@ -1,7 +1,225 @@
...
+];
+
+const themes = [
+	{ label: 'Light', value: 'light' },
+	{ label: 'Dark', value: 'dark' },
+];
+
+const settingsValidationSchema = Yup.object().shape({
+	theme: Yup.string().required('Theme setting is required'),
+	username: Yup.string().required('Username is required').min(3, 'Username must be at least 3 characters'),
```
Comment (ronaldomaiacorrea) [inline client/src/features/settings/Settings.tsx:31]: All constants should go to a separate file inside the folder *lib*.
This is the format of the constants: MINIMUN_USER_NAME_SIZE for example

File: client/src/features/settings/Settings.tsx [hunk 3]
```diff
@@ -1,7 +1,225 @@
...
+	password: Yup.string()
+		.min(8, 'Password must be at least 8 characters')
```
Comment (ronaldomaiacorrea) [inline client/src/features/settings/Settings.tsx:33]: Same here

**Review by ronaldomaiacorrea — CHANGES_REQUESTED** [review]
In the Description of the PR you have to include all the states of the form, so it can be judged. For example, when the fields are not valid, the confirmation of login and whatever you think it is important to be reviewed.

File: client/src/components/PrivateRoute.tsx (hunk 1 above)
Comment (fernandorochaworld) [reply client/src/components/PrivateRoute.tsx]: moved

**Review by ronaldomaiacorrea — APPROVED** [review]

**Pull request merged**